lib/biker.rb

```diff
@@ -1,3 +1,43 @@
 class Biker
+    attr_reader :name,
+                :max_distance,
+                :rides,
+                :acceptable_terrain
+
+    def initialize(name, max_distance)
+        @name = name
+        @max_distance = max_distance
+        @rides = {}
+        @acceptable_terrain = []
+    end
+
+    def learn_terrain!(terrain)
+        @acceptable_terrain << terrain
+    end
+
+    def log_ride(ride, ride_distance)
+       if @rides.key?(ride)
+        @rides[ride] << ride_distance
+       else
+        @rides[ride] = [ride_distance]
+       end
+        # binding.pry
+    end
+
+    def personal_record(ride)
+       return nil unless @rides.has_key?(ride)
+        @rides[ride].min               
+    end
+
+    # def knows_terrain_and_can_log
+    #     if @biker.learn_terrain! 
```

Comment:
Looking at your comments, you might have better luck checking terrain in the log_ride method using the @acceptable_terrain array rather than the learn_terrain! method